### ISSUE TITLE
Restarting node 1.4.4 cluster resulted in red state, with dangling indices

### ISSUE THREAD

**Issue by rtoma** [elastic/elasticsearch#10073]
Hi,

How to recover from the following?
```
dangling index, exists on local file system, but not in cluster metadata,
auto import to cluster state [YES]
```

This morning (~6 hours ago) we restarted one node (not master). After restart the following was in the logging. Cluster is now in red state, because primary and replica shards are unassigned.

```
[2015-03-12 11:49:44,001][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-adm-log4json-2015.01.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,002][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-shd-apache-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,004][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-adm-apache-2015.01.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,004][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-pro-log4j-2015.01.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,005][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-squid-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,005][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-apache-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,006][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-shd-log4json-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,007][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-shd-squid-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,007][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-log4json-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,008][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-log4j-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,008][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-endeca-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,009][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-pro-endeca-2015.01.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,009][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] [logstash-xpr-oracle-2015.02.26] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state [YES]
[2015-03-12 11:49:44,085][INFO ][http                     ] [adm-logsearch-db-001.esprodata01] bound_address {inet[/10.98.252.21:9200]}, publish_address {inet[/10.98.252.21:9200]}
[2015-03-12 11:49:44,085][INFO ][node                     ] [adm-logsearch-db-001.esprodata01] started
```

Please advice.

**Comment by s1monw** [comment]
please use the mailing list for questions like this. this is only for dev issues/bug. 

Regarding your question, if you don't use those indices why don't you just delete them?

**Comment by rtoma** [comment]
Hi Simon,

I guess I used to much 'I need support' trigger words?

My issue was: holy crap, ES is dropping indices after a restart of a single node. This sounds like a bug right?


Anyways I've done some more digging and think I have found something interesting for you. It seems that the dangling indices got deleted by curator. Some 7 hours later one node got restarted and the deleted indices reappeared, but without data, so cluster got red. Thats a bug to me.


Lets take one of those dangling indices as example: logstash-xpr-apache-2015.02.26

Here is what happens:
- curator deletes it at Mar 12th 04:05
- 1 node is restarted and upon startup at Mar 12th 11:49 it detected a dangling index
- the master auto-imports it at Mar 12th 11:49
- cluster got red, because the dangling indices did not have a primary or replica available
- curator deletes it (again) at Mar 13th 04:07
- cluster goes green

So on 1st DELETE did work only half: data got removed, but index stayed in local metadata on 001.


Raw logs:
```
# on master
[2015-03-12 04:05:39,636][INFO ][cluster.metadata         ] [shd-logsearch-db-007.esprodata07] \
  [logstash-xpr-apache-2015.02.26] deleting index

# on node 001
[2015-03-12 11:49:44,005][INFO ][gateway.local.state.meta ] [adm-logsearch-db-001.esprodata01] \
  [logstash-xpr-apache-2015.02.26] dangling index, exists on local file system, but not in cluster 
  metadata, auto import to cluster state [YES]

# on master
[2015-03-12 11:49:44,103][INFO ][gateway.local.state.meta ] [shd-logsearch-db-007.esprodata07] \
  auto importing dangled indices  ... [logstash-xpr-apache-2015.02.26/OPEN]... ] from \
  [[adm-logsearch-db-001.esprodata01][568FE5YIQDqOLBA_JJbOZw][adm-logsearch-db-\
  001.bolcom.net][inet[/10.98.252.21:9300]]{datacenter=ams5, zone=adm-logsearch-db-001, \
  master=true}]
```

**Comment by rtoma** [comment]
FYI, restarted the node 001 again and now no more zombie indices.

**Comment by s1monw** [comment]
@rtoma sorry for moving so quickly. If you stop a node, delete indices and start that node after that again you will get those indices back. That is just a safety feature to not wipe your data without asking you to do it. I am afraid that this is a feature rather than a bug. It's just safer to bring them back and let the user decide. you can opt out of this feature but in 2.0 we make it non-opt-out and always import. Hope this makes sense?

**Comment by rtoma** [comment]
@s1monw not a problem. I seem to be unable to relay the issue. 
Its not:
- stop
- delete
- start
- issue

But:
- delete
- stop
- start
- issue

So this is not the dangling issue, but more that an index delete did delete its data, but not its metadata on disk.

**Comment by s1monw** [comment]
yeah we fixed several things related to this in 1.5 that makes sure we delete the metadata even if the index data can not be deleted. We should likely go further and write tombstones instead of deleting to make sure even if we can't delete we mark indices as deleted
